src/compojure/api/api.clj

```diff
@@ -50,14 +52,19 @@
   [& body]
   (let [[options handlers] (common/extract-parameters body false)
         options (rsc/deep-merge api-defaults options)
-        handler (apply c/routes (concat [(swagger/swagger-routes (:swagger options))] handlers))
+        raw (:raw options)
+        swagger-ui (-> (swagger/swagger-routes (:swagger options))
+                       (cond-> raw ;; if the api won't be wrapped, swagger-ui still needs to be
```

Comment:
but in the case you disable the whole `api-middleware`, you should have your own JSON-middleware setup outside of the `api` anyway to get any data out? Also, if you set your formats to `nil` and disable the `api-middleware`, this wrapping doesn't do the serialization...